mobly/controllers/android_device_lib/snippet_client.py

```diff
@@ -32,42 +31,40 @@ class Error(Exception):
 
 
 class SnippetClient(jsonrpc_client_base.JsonRpcClientBase):
-    def __init__(self, package, port, adb_proxy):
+    def __init__(self, package, host_port, adb_proxy):
         """Initialzies a SnippetClient.
   
         Args:
           package: (str) The package name of the apk where the snippets are
             defined.
-          port: (int) The port at which to start the snippet client. Note that
-            the same port will currently be used for both the device and host
-            side of the connection.
+          host_port: (int) The port at which to start the snippet client. Note
+            that the same port will currently be used for both the device and
+            host side of the connection.
             TODO(adorokhine): allocate a distinct free port for both sides of
             the connection; it is not safe in general to reuse the same one.
           adb_proxy: (adb.AdbProxy) The adb proxy to use to start the app.
         """
-        super(SnippetClient, self).__init__(adb_proxy)
+        super(SnippetClient, self).__init__(host_port, adb_proxy)
         self.app_name = package
-        self._package = package
-        self._port = port
-
-    @property
-    def package(self):
-        return self._package
+        # TODO(adorokhine): Don't assume that a free host-side port is free on
+        # the device as well. Both sides should allocate a unique port.
+        self.device_port = host_port
+        self.package = package
```

Comment:
This is python; anything can be changed...

properties are not really used for access control. They are intended for when a value has nontrivial computation, needs to be cached, etc. 

Here we don't need to compute this value since we already have it, so it should just be a simple public attribute.

http://stackoverflow.com/questions/1554546/when-and-how-to-use-the-builtin-function-property-in-python
http://blaag.haard.se/What-s-the-point-of-properties-in-Python/